scipy testsuite fails – package missing

```
_________________ ERROR collecting datasets/tests/test_data.py _________________
ImportError while importing test module '/sage/local/var/lib/sage/venv-python3.11/lib/python3.11/site-packages/scipy/datasets/tests/test_data.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../../../lib/python3.11/site-packages/scipy/datasets/tests/test_data.py:10: in <module>
    import pooch
E   ModuleNotFoundError: No module named 'pooch'

During handling of the above exception, another exception occurred:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
../../../../../lib/python3.11/site-packages/scipy/datasets/tests/test_data.py:12: in <module>
    raise ImportError("Missing optional dependency 'pooch' required "
E   ImportError: Missing optional dependency 'pooch' required for scipy.datasets module. Please use pip or conda to install 'pooch'.
=========================== short test summary info ============================
ERROR datasets/tests/test_data.py
```

Related:
- https://github.com/scipy/scipy/pull/17965
- https://github.com/scipy/scipy/pull/17937
- https://github.com/scipy/scipy/issues/17710

--- a/build/pkgs/scipy/patches/tests_no_internet_dependency.patch
+++ b/build/pkgs/scipy/patches/tests_no_internet_dependency.patch
@@ -1,0 +1,12 @@
+Disable tests in scipy/datasets, which require internet access.
+See https://github.com/scipy/scipy/pull/17965
+
+diff --git a/scipy/datasets/meson.build b/scipy/datasets/meson.build
+index 10137725381..132e8d29534 100644
+--- a/scipy/datasets/meson.build
++++ b/scipy/datasets/meson.build
+@@ -11,4 +11,3 @@ py3.install_sources(
+   subdir: 'scipy/datasets'
+ )
+ 
+-subdir('tests')

